### DISCUSSION TITLE
Using NuGet.Build.Tasks.Pack in a non-SDK-Style project, why is this dependency added to the created Nuget Package?

### DISCUSSION THREAD

**Discussion by lgsndbox** [dotnet/msbuild#7556] [Q&A]
Hello,

I'm way over my head here, so my question is surely very stupid.
In non-SDK-Style projects, a reference to the package NuGet.Build.Tasks.Pack must be added.
I was naively thinking this was necessary for the package creation to work, but I see that it is added as a dependency to the created Nuget package.

Is it normal?

Thanks.

**Comment by Nirmal4G** [comment]
1. If you're using `packages.config`, You should add it as a development dependency.
2. If you're using `PackageReference` set the `PrivateAssets="All"` or `IncludeAssets="Build;BuildMultiTargeting"` and `ExcludeAssets="All"`

@zivkan please confirm this! and correct me if I'm wrong!

**Reply by zivkan** [reply]
NuGet.Build.Tasks.Pack shouldn't be used for packages.config projects. packages.config projects should only use nuget.exe pack, or migrate to PackageReference.

For non-SDK style projects using PackageReference, if installing the package via Visual Studio (package manager UI, or package manager console), NuGet/Visual Studio will automatically create the PackageReference like this:

```xml
    <PackageReference Include="NuGet.Build.Tasks.Pack">
      <Version>6.2.1</Version>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
      <PrivateAssets>all</PrivateAssets>
    </PackageReference>
```

The `IncludeAssets` are unnecessary, and arguably `runtime` assets for a development dependency is just wrong. But in any case when packing the project like this, it won't become a dependency of the package.

If you're hand-editing the csproj, then this is the quickest, easiest syntax: `<PackageReference Include="NuGet.Build.Tasks.Pack" Version="6.2.1" PrivateAssets="all" />`.
